src/printer.ts

```diff
@@ -1295,12 +1295,27 @@ export class PugPrinter {
 
 		if (this.previousToken?.type === 'dot') {
 			const lastTagToken: TagToken | undefined = previousTagToken(this.tokens, this.currentIndex);
+			let typeAttributeToken: AttributeToken | undefined;
+			if (lastTagToken != null) {
+				const tokens: Token[] = this.tokens.slice(this.tokens.indexOf(lastTagToken) + 1, this.currentIndex - 1);
+				typeAttributeToken = tokens.find((token) => token.type === 'attribute' && token.name === 'type') as
+					| AttributeToken
+					| undefined;
+			}
 
 			let parser: BuiltInParserName | undefined;
 			switch (lastTagToken?.val) {
-				case 'script':
-					parser = 'babel';
+				case 'script': {
+					const knownJSTypeValues: string[] = [
+						"'text/javascript'",
+						"'application/javascript'",
+						"'application/ecmascript'"
+					];
+					if (typeAttributeToken == null || knownJSTypeValues.includes(typeAttributeToken.val as string)) {
+						parser = 'babel';
+					}
```

Comment:
@OiYouYeahYou Do you think this is sufficient?
Also I could add `'application/ld-json'` and apply a json formatter.
But I don't know all possible json memetypes :thinking:
Also it would remove some unnecessary quotes and format tabs to spaces if it configured this way. IMO this could be a good thing.

Reply:
I would replace the array with a map `type` -> `parser` that can be extended overwritten by the config.
````js
{
default: "babel",
"application/json": "babel",
"application/java": "checkstyle",...
}
````
That way you only have to care about the well known ones and the user is still able to use their exotic ones.

Reply:
Maybe with an explicit `none` to disable formatting for the element.

Reply:
@OiYouYeahYou nevermind, @ST-DDT hinted me to an improved solution and as it now uses the json parser instead of json in js it also preserve the quotes due to json keys are always quoted
Haven't thought about this yesterday :slightly_smiling_face: